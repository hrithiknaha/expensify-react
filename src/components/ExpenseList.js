import React from 'react';
import { connect } from 'react-redux'

import ExpenseListItem from './ExpenseListItems'
import selectExpense from '../selectors/expenses'

export const ExpenseList = (props) => {
    return (
        <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses </p>
            ) : (
                props.expenses.map((expense) => {
                    return <ExpenseListItem key={expense.id} {...expense} />        //Sending props as object spread
                })
            )
        }
        </div>
    )
}

const mapStateToProps = (state) => {    //Function which will do the logic inside the connect()     //State coming from app.js, all the files are calling them by importing and exporting therefore it was not clear, now it is
    return {
        expenses: selectExpense(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList)    //Second argument is my wrapped function, or the normal component of a HOC

//As the state changes the components rerenders immediately