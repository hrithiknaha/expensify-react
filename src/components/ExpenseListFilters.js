import React from 'react'
import { connect } from 'react-redux'
import { setTextFilter, sortByDate, sortByAmount, setEndDate, setStartDate } from '../actions/filters'
import { DateRangePicker } from 'react-dates'

class ExpenseListFilters extends React.Component {
    state = {
        calenderFocused: null
    }
    onDatesChange = ({startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate))
        this.props.dispatch(setEndDate(endDate))
    }
    onFocusChange = (focusedInput) => {
        this.setState(() => {
            return {
               calenderFocused: focusedInput 
            }
        })
    }
    render(){
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input placeholder= "Search Expense"className="text-input" type="text" value={this.props.filters.text} onChange = {(e) =>{
                            this.props.dispatch(setTextFilter(e.target.value))
                        }}/>
                    </div>
                    <div className="input-group__item">
                        <select
                        className= "select"
                        value={this.props.filters.sortBy}
                        onChange={(e) => {
                            if(e.target.value === 'date')
                                this.props.dispatch(sortByDate())
                            else if (e.target.value === 'amount')
                                this.props.dispatch(sortByAmount())
                        }}
                        >
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker 
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calenderFocused}
                            onFocusChange={this.onFocusChange}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                            showClearDates={true}
                        />
                    </div>
                </div>  
            </div>
        )
    }
}
const mapStateToProps = (state) =>{
    return {
        filters: state.filters          //Sending as a prop to the above component
    }
}

export default connect(mapStateToProps)(ExpenseListFilters)