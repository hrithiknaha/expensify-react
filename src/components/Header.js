import React from 'react';
import { NavLink } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogout } from '../actions/auth'

export const Header = ({ startLogout }) => {
    return (
        <div>
            <h1>Expensify</h1>
            <NavLink to="/dashboard" activeClassName="is-active" S>Home Page</NavLink>
            <NavLink to="/create" activeClassName="is-active">Create Page</NavLink>
            <button onClick={ startLogout }>Logout</button>
        </div>
    )
}

const mapDispatchToProps = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout())
    }
}

export default connect(undefined, mapDispatchToProps)(Header)