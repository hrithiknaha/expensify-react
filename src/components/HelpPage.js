import React from 'react';

const HelpPage = () =>{
    return (
        <div>
            This is from my page help
        </div>
    )
}

export default HelpPage;