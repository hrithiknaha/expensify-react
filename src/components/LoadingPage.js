import React from 'react'

const LoadingPage = () => {
    return (
        <div className="loader">
            <img className="loader__image" src="/images/5.gif" alt=""/>
        </div>
    )
}

export default LoadingPage