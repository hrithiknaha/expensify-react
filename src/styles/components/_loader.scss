.loader {
    height: 100vh;
    width: 100vw;
    display: flex;
    justify-content: center;
    align-items: center;
}

.loader__image{
    height: 6rem;
    widows: 6rem;
}