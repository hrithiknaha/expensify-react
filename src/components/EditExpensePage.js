import React from 'react';
import ExpenseList from './ExpenseList';
import { connect } from  'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from  '../actions/expenses'

const EditExpensePage = (props) =>{
    return (
        <div>
            <ExpenseForm 
                expense={props.expense}
                onSubmit ={(expense) =>{
                    props.dispatch(startEditExpense(props.expense.id, expense))
                    props.history.push('/');
                }}
            />
            <button onClick={() => {
                props.dispatch(startRemoveExpense({id: props.expense.id}))       //Dispatch is also send as a prop, hence we can use it directly
                props.history.push('/');
            }}>Remove</button>
        </div>
    )
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id
        })
    }
}

export default connect(mapStateToProps)(EditExpensePage);