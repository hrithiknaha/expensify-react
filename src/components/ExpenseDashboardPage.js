import React from 'react';
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummary from './ExpenseSummary'

const ExpenseDashboardPage = () =>{
    return (
        <div>
            <ExpensesSummary />
            <ExpenseList />
            <ExpenseListFilters />
        </div>
    )
}

export default ExpenseDashboardPage;