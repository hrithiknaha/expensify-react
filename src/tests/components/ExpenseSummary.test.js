import React from 'react'
import {shallow} from 'enzyme'
import {ExpensesSummary} from '../../components/ExpenseSummary'

test('should correctly render ExpenseSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={235} />)
    expect(wrapper).toMatchSnapshot();
})

test('should correctly render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expenseTotal={23532432434} />)
    expect(wrapper).toMatchSnapshot();

})