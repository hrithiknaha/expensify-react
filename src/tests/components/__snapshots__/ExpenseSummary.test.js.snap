// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;
